cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -1293,12 +1295,14 @@ std::shared_ptr<const KeyValueMetadata> GetObjectMetadata(const ObjectResult& re
 
 template <typename ObjectRequest>
 struct ObjectMetadataSetter {
+  static constexpr std::string_view CONTENT_TYPE_KEY = "Content-Type";
```

Comment:
Nit: naming conventions for constants
```suggestion
  static constexpr std::string_view kContentTypeKey = "Content-Type";
```